src/components/App/App.tsx

```diff
@@ -126,50 +131,58 @@ class App extends React.Component<Props> {
-                favorites={this.state.favorites}
-              />
-            </main>
-          )
-        }}/>
-        <Route
-          path="/:title"
-          render={({ match }) => {
-            const data = this.state.apiList.find((api) => api.API === match.params.title);
-            if (data) {
-              const myNotes = this.state.savedNotes.find(savedNote => savedNote.name === data.API)
-              const savedNotes = myNotes?.notes.map((note, index) => {
-                return <Note note={note} key={index} apiName={myNotes.name} deleteNote={this.deleteNote} />
-              })
-              return (
+    if (this.state.error) {
+      return <ErrorMessage statusCode={this.state.error} />
+    } else {
+      return (
+        <div className="App">
+          <SideBar></SideBar>
+          <Switch>
+          <Route exact path='/' render={() => {
+            return (              
                 <main>
-                  <FeaturedCard 
-                    {...data} 
+                  <h1>metAPI</h1>
+                  <FilterForm filter={this.filter} apiList={this.state.apiList}/>
+                  <CardContainer 
+                    apiList={this.state.currentApis} 
                     toggleFavorite={this.toggleFavorite} 
-                    saveNote={this.saveNote} 
-                    favorites={this.state.favorites} 
+                    favorites={this.state.favorites}
                   />
-                  <section className='saved-notes'>
-                    <h3>Notes</h3>
-                    {savedNotes}
-                  </section>
                 </main>
-              );
-            }
-          }}
-        />
-      </div>
-    );
+            )}}
+          />
+          <Route
+            exact path="/api/:title"
+            render={({ match }) => {
+              console.log(match)
+              const data = this.state.apiList.find((api) => api.API === match.params.title);
+              if (data) {
+                const myNotes = this.state.savedNotes.find(savedNote => savedNote.name === data.API)
+                const savedNotes = myNotes?.notes.map((note, index) => {
+                  return <Note note={note} key={index} apiName={myNotes.name} deleteNote={this.deleteNote} />
+                })
+                return (
+                    <main>
+                      <FeaturedCard 
+                        {...data} 
+                        toggleFavorite={this.toggleFavorite} 
+                        saveNote={this.saveNote} 
+                        favorites={this.state.favorites} 
+                      />
+                      <section className='saved-notes'>
+                        <h3>Notes</h3>
+                        {savedNotes}
+                      </section>
+                    </main>
+                );
+              }
+            }}
+          />
+          <Route render={() => <ErrorMessage statusCode={"404"}/>} />
```

Comment:
Sweet, definitely think a switch like this is the best for an error page like this